homeassistant/components/ambient_network/entity.py

```diff
@@ -0,0 +1,52 @@
+"""Base entity class for the Ambient Weather Network integration."""
+from __future__ import annotations
+
+from abc import abstractmethod
+
+from homeassistant.core import callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import AmbientNetworkDataUpdateCoordinator
+
+
+class AmbientNetworkEntity(CoordinatorEntity[AmbientNetworkDataUpdateCoordinator]):
+    """Entity class for Ambient network devices."""
+
+    _attr_attribution = "Data provided by ambientnetwork.net"
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: AmbientNetworkDataUpdateCoordinator,
+        description: EntityDescription,
+        station_name: str,
+    ) -> None:
+        """Initialize the Ambient network entity."""
+
+        super().__init__(coordinator=coordinator)
+        self.entity_description = description
+        self.station_name = station_name
+        self._device_id = station_name
+        self._attr_unique_id = f"{station_name}_{description.key}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, station_name)},
```

Comment:
Shouldn't the Mac address be a better identifier for the station?

Reply:
Yes. Fixed.